### PR TITLE
disallow ICA upgrades that change channel order + increase test coverage

### PR DESCRIPTION
**Pull request by crodriguezvega**
## Description

This PR:
- adds tests to increase coverage for upgrade callbacks in ICA.
- Adds check to disallow upgrades that change the channel ordering.

closes: #XXXX

### Commit Message / Changelog Entry

```text
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/apps/27-interchain-accounts/host/ibc_module_test.go [hunk 1]
```diff
@@ -603,6 +604,82 @@ func (suite *InterchainAccountsTestSuite) TestOnTimeoutPacket() {
...
+				path.EndpointB.ChannelID,
+				order,
+				[]string{path.EndpointB.ConnectionID},
+				path.EndpointA.ChannelConfig.ProposedUpgrade.Fields.Version,
+			)
+
+			if tc.expPass {
+				suite.Require().NoError(err)
+			} else {
+				suite.Require().Error(err)
```
Comment (colin-axner) [inline modules/apps/27-interchain-accounts/host/ibc_module_test.go:676]: super nit: would be nice to use `suite.Require().ErrorIs()`

File: modules/apps/27-interchain-accounts/controller/ibc_middleware_test.go [hunk 2]
```diff
@@ -762,6 +764,201 @@ func (suite *InterchainAccountsTestSuite) TestOnTimeoutPacket() {
...
+					order,
+					[]string{path.EndpointA.ConnectionID},
+					version,
+				)
+			}
+
+			if tc.expPass {
+				suite.Require().NoError(err)
+			} else {
+				suite.Require().Error(err)
```
Comment (colin-axner) [inline modules/apps/27-interchain-accounts/controller/ibc_middleware_test.go:863]: same here for `suite.Require().ErrorIs`

File: modules/apps/27-interchain-accounts/controller/ibc_middleware_test.go [hunk 3]
```diff
@@ -762,6 +764,201 @@ func (suite *InterchainAccountsTestSuite) TestOnTimeoutPacket() {
...
+			module, _, err := suite.chainA.App.GetIBCKeeper().PortKeeper.LookupModuleByPort(suite.chainA.GetContext(), path.EndpointA.ChannelConfig.PortID)
+			suite.Require().NoError(err)
+
+			app, ok := suite.chainA.App.GetIBCKeeper().Router.GetRoute(module)
+			suite.Require().True(ok)
+			cbs, ok := app.(porttypes.UpgradableModule)
+			suite.Require().True(ok)
+
+			if isNilApp {
+				cbs = controller.NewIBCMiddleware(nil, suite.chainA.GetSimApp().ICAControllerKeeper)
```
Comment (colin-axner) [inline modules/apps/27-interchain-accounts/controller/ibc_middleware_test.go:848]: I don't understand the purpose of this if / else. Where are the cbs invoked?

**Review by colin-axner — APPROVED** [review]
ACK changes, but see testing suggestions

File: modules/apps/27-interchain-accounts/controller/ibc_middleware_test.go [hunk 4]
```diff
@@ -18,8 +18,11 @@ import (
 	porttypes "github.com/cosmos/ibc-go/v8/modules/core/05-port/types"
 	host "github.com/cosmos/ibc-go/v8/modules/core/24-host"
 	ibctesting "github.com/cosmos/ibc-go/v8/testing"
+	ibcmock "github.com/cosmos/ibc-go/v8/testing/mock"
 )
 
+const invalidVersion = "invalid|version"
```
Comment (damiannolan) [inline modules/apps/27-interchain-accounts/controller/ibc_middleware_test.go:24]: I think there might be similar constants in the `ibctesting` pkg

**Review by damiannolan — APPROVED** [review]

File: modules/apps/27-interchain-accounts/controller/ibc_middleware_test.go (hunk 4 above)
Comment (crodriguezvega) [reply modules/apps/27-interchain-accounts/controller/ibc_middleware_test.go:24]: I cannot find one, but if you know where it is, I am glad to change this in a follow-up PR.

Comment (damiannolan) [reply modules/apps/27-interchain-accounts/controller/ibc_middleware_test.go:24]: I think some other tests use `InvalidID` to chuck into the version, but happy to keep it as is

**Pull request merged**